cli/src/main/java/com/liferay/blade/cli/ExtensionsClassLoaderSupplier.java

```diff
@@ -137,15 +139,54 @@ private void _extractBladeExtensions(Path extensionsDirectory) throws IOExceptio
 			Set<Object> keySet = properties.keySet();
 
 			ClassLoader classLoader = Extensions.class.getClassLoader();
+			
 
-			for (Object key : keySet) {
-				String extension = key.toString() + "-" + properties.getProperty(key.toString()) + ".jar";
+			try {
+				Set<String> extensions = new HashSet<>();
+				
+
+				for (Object key : keySet) {
+					String extension = key.toString() + "-" + properties.getProperty(key.toString()) + ".jar";
+					
 
-				try (InputStream extensionInputStream = classLoader.getResourceAsStream(extension)) {
-					Path extensionPath = extensionsDirectory.resolve(extension);
+					if (classLoader.getResource(extension) != null) {
+						extensions.add(extension);
+					}
+					else {
+						String errorString = String.format("Unable to locate %s on classpath", extension);
 
-					Files.copy(extensionInputStream, extensionPath, StandardCopyOption.REPLACE_EXISTING);
+						throw new NoSuchElementException(errorString);
+					}
 				}
+				
+
+				for (String extension : extensions) {
```

Comment:
Lets have two try/catch blocks.  one that covers all this code (like you have here) and another try/catch for indiviidual extensions, incase there is a bad actor (uncopiable extension file) or something, but having two levels of try/catch would let us load more etensions if there is a bad one in the list.